### PR TITLE
feat: use new field filter for /api/metadata

### PR DESCRIPTION
**Pull request by mortenoh**
Updates metadata endpoints to use the new field filtering service.

* `/api/metadata`
* `/api/dataSets/X/metadata`
* `/api/programs/X/metadata`

This also means XML is _not_ supported on these endpoints anymore and only JSON will be returned.

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java [hunk 1]
```diff
@@ -129,6 +121,29 @@ private ObjectMapper configureFieldFilterObjectMapper( ObjectMapper objectMapper
         return objectNodes;
     }
 
+    public ObjectNode createObjectNode()
```
Comment (jbee) [inline dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java:124]: `private`?

**Review by jbee — APPROVED** [review]
Just a minor comment.

File: dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java (hunk 1 above)
Comment (mortenoh) [reply dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java:124]: I'll probably move them at some point, but right now these are utilty methods used outside of this class for constructing JSON objects and arrays.

**Review by larshelge — APPROVED** [review]

**Comment by larshelge** [comment]
Looks great to me.

**Pull request merged**